### ISSUE TITLE
Decoding the words using tokens of same language on an ASR trained using a code-switched bilingual dataset

### ISSUE THREAD

**Issue by manjuke** [NVIDIA/NeMo#5458]
Hi All,
I have trained a Hindi-English code-switched ASR system using the dataset having the transcription written in English (for English words) &  Hindi (for Hindi words). 
Example:
 क्या सौ रुपए loan लेना चाहती है आप
The tokenization script resulted in all the letters from Hindi + English for vocabulary creation. Nemo pretrained Hindi model followed by transfer learning using this dataset is done. When I decode on test set using this model I get some words having tokens from both languages.
Example: "doubल"  --> "doub" (English) + "ल" (Hindi) --> ideally it should have decoded as "double".. 

I want to know, is there any way (any settings/options) to restrict all the tokens in a word to be from a single language (and not a mix of more than one language). Please suggest. 
Thanks

**Comment by manjuke** [comment]
Thank you
